### PR TITLE
Guard ensureScrollValueMonitoring against popup blockers overriding document.createEvent()

### LINKED ISSUE #6887: ensureScrollValueMonitoring issue
By having our app wrapped with Sentry (error catching service) we've encountered a bug - which may not be React specific but some kind of a workaround for it should be done. With that in mind I'm not sure how to tackle it yet.

https://github.com/facebook/react/blob/1573baaee8ba1c9cd84145b01effcb5eb80503fb/src/renderers/dom/client/ReactBrowserEventEmitter.js#L363

`TypeError: Cannot use 'in' operator to search for 'pageX' in null` - which occurs during mounting a root component into the DOM (using React 15)

It happens really rarely in Chrome (so far caught only there, on versions 49 and 50, both Mac and Windows). With thousands renders of our app daily we got only 7 such hits in the past 3 days.

I've tried to search any info how `document.createEvent('MouseEvent')` can return `null`, but couldn't find any viable info.

With React's popularity is odd too me that nobody has filled this issue earlier which leds me to think that maybe our usage is somehow different than others and the only thing 'kinda' special is that our app is running inside an iframe.

### PR DESCRIPTION
**Pull request by Andarist**
Fixes #6887

cc @gaearon

### CODE REVIEW

File: src/renderers/dom/client/__tests__/ReactBrowserEventEmitter-test.js [hunk 1]
```diff
@@ -452,4 +452,18 @@ describe('ReactBrowserEventEmitter', function() {
...
+    document.createEvent = function() {
+      return null;
+    };
+    spyOn(document, 'createEvent');
+
+    var hasEventPageXY = ReactBrowserEventEmitter.testEventPageXY();
+    expect(document.createEvent.calls.count()).toBe(1);
+    expect(hasEventPageXY).toBe(false);
+
+    document.createEvent = originalCreateEvent;
```
Comment (gaearon) [inline src/renderers/dom/client/__tests__/ReactBrowserEventEmitter-test.js]: Could you please wrap expectations into `try / finally` so `document.createEvent` is always restored regardless of whether the test fails?

File: src/renderers/dom/client/ReactBrowserEventEmitter.js [hunk 2]
```diff
@@ -345,6 +345,19 @@ var ReactBrowserEventEmitter = Object.assign({}, ReactEventEmitterMixin, {
   },
 
   /**
+   * Protect against document.createEvent() returning null
+   * Some popup blocker extensions appear to do this:
+   * https://github.com/facebook/react/issues/6887
+   */
+  testEventPageXY: function() {
```
Comment (gaearon) [inline src/renderers/dom/client/ReactBrowserEventEmitter.js]: Maybe let’s call it `supportsEventPageXY()` so it’s clear what’s being returned.

**Comment by Andarist** [comment]
@gaearon amended along your suggestions

**Comment by gaearon** [comment]
Thanks.

**Pull request merged**